src/Altinn.App.Core/Expressions/LayoutEvaluatorState.cs

```diff
@@ -0,0 +1,193 @@
+using Altinn.App.Core.Configuration;
+using Altinn.App.Core.Helpers;
+using Altinn.App.Core.Models.Expressions;
+using Altinn.App.Core.Models.Layout;
+using Altinn.App.Core.Models.Layout.Components;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Expressions;
+
+/// <summary>
+/// Collection class to hold all the shared state that is required for evaluating expressions in a layout.
+/// </summary>
+public class LayoutEvaluatorState
+{
+    private readonly IDataModelAccessor _dataModel;
+    private readonly LayoutModel _componentModel;
+    private readonly FrontEndSettings _frontEndSettings;
+    private readonly Instance _instanceContext;
+
+    /// <summary>
+    /// Constructor for LayoutEvaluatorState. Usually called via <see cref="LayoutEvaluatorStateInitializer" /> that can be fetched from dependency injection.
+    /// </summary>
+    public LayoutEvaluatorState(IDataModelAccessor dataModel, LayoutModel componentModel, FrontEndSettings frontEndSettings, Instance instance)
+    {
+        _dataModel = dataModel;
+        _componentModel = componentModel;
+        _frontEndSettings = frontEndSettings;
+        _instanceContext = instance;
+        // TODO: shaddowFields ...
```

Comment:
I guess this is for later when we actually implement shadow fields so we can just delete the TODO for now.